python/tests/unit/test_cache.py

```diff
@@ -71,7 +71,7 @@ def apply_yaml_string(self, yaml_data: str, allow_updates=True) -> None:
             name = metadata['name']
             namespace = metadata['namespace']
 
-            key = f"{kind}-v2-{name}-{namespace}"
```

Comment:
Just leaving a comment here as a mental note that my changes to the cache for the regression around cache keys will need to be checked before we repatriate 2.3 to ensure they dont overwrite anything here.

Reply:
I will repatriate first and rebase on your changes before officially merging it. That way we have a clean commit. That will also give Luke a chance to give a final review.

Reply:
I have re-based and pushed so this should be resolved...unless you think otherwise?